src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,219 @@
+            </select>
+          </div>
+          <div>
+            <h4>{{ getNGramLabel() }}</h4>
+            <ul>
+              <li v-for="(item, index) in filteredKeywords" :key="index">{{ item }}</li>
+            </ul>
+          </div>
         </div>
+      </div>
     </div>
-</template>
\ No newline at end of file
+  </div>
+</template>
+
+<script setup>
+import { ref, computed, onMounted } from 'vue';
+
+const text = ref('');
+const keywords = ref({
+  unigrams: [],
+  bigrams: [],
+  trigrams: []
+});
+const selectedNGram = ref('unigrams');
+const isKeywordsGenerated = ref(false);
+const textarea = ref(null);
+
+const generateKeywords = () => {
+  const words = cleanText(text.value).split(/\s+/);
+  keywords.value.unigrams = getNGrams(words, 1);
+  keywords.value.bigrams = getNGrams(words, 2);
+  keywords.value.trigrams = getNGrams(words, 3);
+  isKeywordsGenerated.value = true;
+  adjustTextareaHeight();
+};
+
+const cleanText = (text) => {
+  return text.replace(/[^\w\s]|_/g, '').replace(/\s+/g, ' ').toLowerCase();
+};
+
+const getNGrams = (words, n) => {
+  const nGrams = [];
+  for (let i = 0; i <= words.length - n; i++) {
+    const nGram = words.slice(i, i + n).join(' ');
+    if (!nGrams.includes(nGram)) {
+      nGrams.push(nGram);
+    }
+  }
+  return nGrams;
+};
+
+const getNGramLabel = () => {
+  switch (selectedNGram.value) {
+    case 'unigrams': return '1-grams';
+    case 'bigrams': return '2-grams';
+    case 'trigrams': return '3-grams';
+    default: return '';
+  }
+};
+
+const filteredKeywords = computed(() => {
+  return keywords.value[selectedNGram.value];
+});
+
+const filterKeywords = () => {
+  filteredKeywords.value; // Trigger recompute
+};
+
+const adjustTextareaHeight = () => {
+  const el = textarea.value;
+  el.style.height = 'auto';
+  el.style.height = `${el.scrollHeight}px`;
+};
+
+onMounted(() => {
+  adjustTextareaHeight();
+});
+</script>
+
+<style scoped>
+.ma-keywords-generator {
+  max-width: 1200px;
+  margin: auto;
+  padding: 2em;
+  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Arial, sans-serif;
+}
+
+.card {
+  background: #ffffff;
+  border-radius: 20px;
+  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
+  padding: 2em;
+}
+
+.container {
+  display: flex;
+  justify-content: center; /* Center the content horizontally */
+  align-items: flex-start;
+  gap: 3em;
+  width: 100%;
+}
+
+.input-container {
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  gap: 1em;
+  width: 100%;
+  transition: transform 0.3s ease, width 0.3s ease;
+  transform: translateX(0);
+}
+
+.input-container.centered {
+  transform: translateX(0); /* Remove the translateX to keep it centered */
+}
+
+.keywords-output {
+  flex: 1;
+  padding-left: 1em;
+  border-left: 2px solid #e0e0e0;
+  max-height: 70vh;
+  overflow-y: auto;
+}
+
+.ma-header {
+  font-size: 1.75em;
+  font-weight: 600;
+  color: #333;
+  text-align: center;
+  margin: 0;
+}
+
+textarea {
```

Comment:
We should not style html elements directly. In the future, suppose we start using `input` instead of `textarea`, the part for `textarea` is wasted.